gk/main.c

```diff
@@ -352,8 +351,8 @@ gk_process_request(struct flow_entry *fe, struct ipacket *packet,
 	 * 2 for capability renew; 3-63 for requests.
 	 */
 	priority += 3;
-	if (unlikely(priority > PRIORITY_MAX))
-		priority = PRIORITY_MAX;
+	if (unlikely(priority > GK_MAX_REQ_PRIORITY))
+		priority = GK_MAX_REQ_PRIORITY;
```

Comment:
It isn't a big deal in this patch, but it represents well the difficulty that I'm having to review this pull request as a whole: each patch is trying to fix unrelated things all at the same time. The title of this pull request reads as **sol: make all compile-time parameters runtime parameters**, and the fix here is in the GK block.

Breaking the patches into patches of related parameters would simplify a lot the whole review process. Sorry for the extra work. I'm just not smart enough to not get lost in this review.

Reply:
My bad! Sorry for the confusion in breaking things.